src/lib/optimizer/strategy/join_detection_rule.cpp

```diff
@@ -98,8 +106,18 @@ std::optional<JoinDetectionRule::JoinCondition> JoinDetectionRule::_find_predica
        * IMPORTANT: Since we only traversed nodes that do not change the column order when looking for the Predicate,
        * we can do this by the simple range check in _is_join_condition()
        */
-      const auto predicate_left_column_id = predicate_node->column_id();
-      const auto predicate_right_column_id = boost::get<ColumnID>(predicate_node->value());
+      auto predicate_left_column_id = predicate_node->column_id();
+      auto predicate_right_column_id = boost::get<ColumnID>(predicate_node->value());
+
+      // The Predicate refers to a subtree left of the subtree the Cross Join came from
+      if (predicate_left_column_id < column_id_offset || predicate_right_column_id < column_id_offset) {
+        continue;
```

Comment:
Why do we only have to check the left side here?

Reply:
Right side check happens in `::_is_join_condition`